src/api/layers/WMSCapabilitiesParser.class.js

```diff
@@ -140,15 +151,56 @@ export default class WMSCapabilitiesParser {
         )
     }
 
-    getLayerExtent(layerId, layer, projection, ignoreError = true) {
+    _getLayerAttributes(layer, parents, projection, ignoreError = true) {
+        let layerId = layer.Name
+        // Some WMS only have a Title and no Name, therefore in this case take the Title as layerId
+        if (!layerId && layer.Title) {
+            // if we don't have a name use the title as name
+            layerId = layer.Title
+        }
+        if (!layerId) {
+            // Without layerID we cannot use the layer in our viewer
+            const msg = `No layerId found in WMS capabilities for layer in ${this.originUrl.toString()}`
+            log.error(msg, layer)
+            if (ignoreError) {
+                return {}
+            }
+            throw new Error(msg)
+        }
+
+        if (!this.version) {
+            throw new Error(`No WMS version found in Capabilities of ${this.originUrl.toString()}`)
+        }
+        if (!WMS_SUPPORTED_VERSIONS.includes(this.version)) {
+            throw new Error(
+                `WMS version ${this.version} of ${this.originUrl.toString()} not supported`
+            )
+        }
```

Comment:
shouldn't `ignoreError = true` outputs the same behavior as above, meaning returning an empty `{}` instead of a `throw ...`?

Reply:
My idea with the `ignoreError = true` was to return as much as possible, e.g. if a capabilities is valid and only one layer element is invalid, then we still parse it and returns layer objects for all other layers. Here if the version is not valid I cannot return anything. But I see that is not 100% clear, I'll have a look if I can improve the code/logic.

Reply:
Done